### PR TITLE
Replace NormalizeWhitespace with manually generated whitespace

### PR DESCRIPTION
**Pull request by AArnott**
For reasons of terrible perf (https://github.com/dotnet/roslyn/issues/54144), we're replacing `NormalizeWhitespace()` with manually generated whitespace to fix the largest contributor to terrible perf as tracked by #244 and #38.

Sample of manual whitespace generated so far on the simplest test:

```cs
 // ------------------------------------------------------------------------------
 // <auto-generated>
 //     This code was generated by a tool.
 //
 //     Changes to this file may cause incorrect behavior and will be lost if
 //     the code is regenerated.
 // </auto-generated>
 // ------------------------------------------------------------------------------
 
 #pragma warning disable CS1591,CS1573,CS0465,CS0649,CS8019,CS1570,CS1584,CS1658
 namespace Windows.Win32
 {
 	using global::System;
 	using global::System.Diagnostics;
 	using global::System.Runtime.CompilerServices;
 	using global::System.Runtime.InteropServices;
 	using win32 = global::Windows.Win32;
 
 /// <content>
 /// Contains extern methods from "Kernel32.dll".
 /// </content>
 internal static partial class PInvoke
 {
 /// <summary>Retrieves the number of milliseconds that have elapsed since the system was started, up to 49.7 days.</summary>
 /// <returns>The return value is the number of milliseconds that have elapsed since the system was started.</returns>
 /// <remarks>
 /// <para><see href="https://docs.microsoft.com/windows/win32/api//sysinfoapi/nf-sysinfoapi-gettickcount">Learn more about this API from docs.microsoft.com.</see></para>
 /// </remarks>
 [DllImport("Kernel32",ExactSpelling=true)]internal static extern uint GetTickCount();
 }
 }
```

### CODE REVIEW

File: src/Microsoft.Windows.CsWin32/Generator.cs [hunk 1]
```diff
@@ -1667,6 +1678,24 @@ protected virtual void Dispose(bool disposing)
...
+            .WithCloseBraceToken(TokenWithLineFeed(SyntaxKind.CloseBraceToken));
+
+        private static UsingDirectiveSyntax UsingDirectiveWithTrivia(NameSyntax name, NameEqualsSyntax? alias = null) => UsingDirective(
+            usingKeyword: Token(TriviaList(ElasticTab), SyntaxKind.UsingKeyword, TriviaList(ElasticSpace)),
+            staticKeyword: default,
+            alias: alias?.WithEqualsToken(Token(TriviaList(ElasticSpace), SyntaxKind.EqualsToken, TriviaList(ElasticSpace))),
+            name: name,
+            semicolonToken: TokenWithLineFeed(SyntaxKind.SemicolonToken));
+
+        private static SyntaxToken TokenWithSpace(SyntaxKind syntaxKind) => Token(TriviaList(), syntaxKind, TriviaList(ElasticSpace));
```
Comment (sharwell) [inline src/Microsoft.Windows.CsWin32/Generator.cs:1695]: 💡 Avoid elastic for this. It's a hint to the formatter but unnecessary for generation that does not go through the formatter on the way to a string.

**Comment by jnm2** [comment]
@AArnott XML doc comments on types look messed up (on 428710543b78f20eb15c9a007cbce2224e556e3a):

```cs
	namespace RestartManager
	{
			/// <summary>Uniquely identifies a process by its PID and the time the process began.</summary>
			/// <remarks>
			/// <para>The <b>RM_UNIQUE_PROCESS</b> structure can be used to uniquely identify an application in an <a href="https://docs.microsoft.com/windows/desktop/api/restartmanager/ns-restartmanager-rm_process_info">RM_PROCESS_INFO</a> structure or  registered with the Restart Manager session by the <a href="https://docs.microsoft.com/windows/desktop/api/restartmanager/nf-restartmanager-rmregisterresources">RmRegisterResources</a> function.</para>
			/// <para><see href="https://docs.microsoft.com/windows/win32/api//restartmanager/ns-restartmanager-rm_unique_process#">Read more on docs.microsoft.com</see>.</para>
			/// </remarks>
		internal partial struct RM_UNIQUE_PROCESS
		{
			/// <summary>The product identifier (PID).</summary>
			internal uint dwProcessId;
			/// <summary>The creation time of the process. The time is provided as a <b>FILETIME</b> structure that is returned by the <i>lpCreationTime</i> parameter of the <a href="https://docs.microsoft.com/windows/desktop/api/processthreadsapi/nf-processthreadsapi-getprocesstimes">GetProcessTimes</a> function.</summary>
			internal global::System.Runtime.InteropServices.ComTypes.FILETIME ProcessStartTime;
		}
```

**Comment by AArnott** [comment]
Yup. And some statements aren't indented. Most of the feedback I got from roslyn and other folks during this PR was on how unreasonably generous I was in producing pretty syntax. The argument was that this code is hardly read, so it doesn't need to be pretty. I'd like to find a balance though, and get customer feedback (including from you) on where that balance is. 
In the meantime, I needed to move on to more pressing items (like the YAML startup perf problem) so I took it with a few remaining whitespace bugs, that I may or may not ever get around to fixing.

**Comment by AraHaan** [comment]
I recently tested it with generating the enum MINIDUMP_TYPES and it resulted in all of the docs not rendering properly.

I hope that despite that they will render properly when people consume my library with those docs with that enum being part of public api for it.

As such properly spaced docs is a must for my code but only when it's placed on public apis.

**Comment by AArnott** [comment]
@AraHaan, I took a look at `MINIDUMP_TYPE` (it's singular, not plural), and I agree the xml docs are very weirdly formatted, but this doesn't have any impact on the consumption of the APIs. The Intellisense documentation appears to be unaffected:

image

**Comment by AraHaan** [comment]
I see, was that image from inside of vs2019?

**Comment by AArnott** [comment]
Either that, or a VS 2022 preview, I don't remember. But it should be the same either way.

**Pull request merged**